File: LSP+ISP/Java/MarsRoverKata/src/main/java/MarsRoverKata/Missile.java
package MarsRoverKata;

public class Missile extends Projectile {
    public static final int range = 5;

    public Missile(Mars mars) {
        super(mars);

    }

    @Override
    protected int getMaxRange() {
        return range;
    }

    @Override
    protected boolean isCollisionDetected(Point desired) {
        Point newDestination = desired;
        newDestination = calculatePositionY(desired, newDestination);
        newDestination = calculatePositionX(desired, newDestination);
        IObstacle obstacle = findObstacle(newDestination);

        return obstacle != null && obstacle.isDestructable();
    }
}

File: DIP/Solution/MarsRoverKata/Missile.cs
namespace MarsRoverKata
{
    public class Missile : Projectile
    {
        public const int Range = 5;

        public Missile(Mars mars)
            : base(mars)
        {
        }

        protected override int MaxRange
        {
            get { return Range; }
        }

        protected override bool IsCollisionDetected(Point desired)
        {
            Point newDestination = desired;
            newDestination = CalculatePositionY(desired, newDestination);
            newDestination = CalculatePositionX(desired, newDestination);
            var obstacle = FindObstacle(newDestination);

            return obstacle != null && obstacle.IsDestructable;
        }
    }
}
